File: app/src/main/java/com/sumian/sd/event/EventBusUtil.java
package com.sumian.sd.event;

import org.greenrobot.eventbus.EventBus;

/**
 * <pre>
 *     @author : Zhan Xuzhao
 *     e-mail : xuzhao.z@sumian.com
 *     time   : 2018/6/7 14:46
 *     desc   :
 *     version: 1.0
 * </pre>
 */
public class EventBusUtil {
    public static void postEvent(Object event) {
        EventBus.getDefault().post(event);
    }

    public static void postStickyEvent(Object event) {
        EventBus.getDefault().postSticky(event);
    }

    public static void removeStickyEvent(Object event) {
        EventBus.getDefault().removeStickyEvent(event);
    }

    public static void register(Object subscriber) {
        EventBus.getDefault().register(subscriber);
    }

    public static void unregister(Object subscriber) {
        EventBus.getDefault().unregister(subscriber);
    }
}

File: doctor/src/main/java/com/sumian/sddoctor/util/EventBusUtil.kt
package com.sumian.sddoctor.util

import org.greenrobot.eventbus.EventBus

/**
 * <pre>
 *     @author : Zhan Xuzhao
 *     e-mail : xuzhao.z@sumian.com
 *     time   : 2018/6/13 18:16
 *     desc   :
 *     version: 1.0
 * </pre>
 */
class EventBusUtil {
    companion object {
        fun postEvent(event: Any) {
            EventBus.getDefault().post(event)
        }

        fun postStickyEvent(event: Any) {
            EventBus.getDefault().postSticky(event)
        }

        fun removeStickyEvent(event: Any) {
            EventBus.getDefault().removeStickyEvent(event)
        }

        fun register(subscriber: Any) {
            EventBus.getDefault().register(subscriber)
        }

        fun unregister(subscriber: Any) {
            EventBus.getDefault().unregister(subscriber)
        }
    }
}
